healpy/visufunc.py

```diff
@@ -1329,14 +1360,16 @@ def azeqview(
             badcolor=badcolor,
             bgcolor=bgcolor,
             norm=norm,
+            alpha=alpha,
         )
         if cbar:
             im = ax.get_images()[0]
             b = im.norm.inverse(np.linspace(0, 1, im.cmap.N + 1))
             v = np.linspace(im.norm.vmin, im.norm.vmax, im.cmap.N)
+            mappable = plt.cm.ScalarMappable(norm=matplotlib.colors.Normalize(vmin=im.norm.vmin, vmax=im.norm.vmax), cmap=cmap)
```

Comment:
why do you need to build the `ScalarMappable` object? how is it related to the alpha channel?

Reply:
@aasensio can you please explain why you need `ScalarMappable`?

Reply:
Sorry, had a busy couple of weeks. Colorbars in `plt.imshow` cannot be used when the alpha channel per pixel is different,  because the colormap is not properly defined. This `ScalarMappable` defines the range of colors of the image to build the colorbar as if there were no alpha channel and it is a workaround this issue with `plt.imshow`.